src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.Paint.cs

```diff
@@ -228,26 +210,15 @@ private void OnSurfacePainted(object? sender, SKPaintSurfaceEventArgs e)
 		_shadowPropertyChanged = false;
 	}
 
-	private bool ProcessContentBackgroundIfNeeded(out Color? contentBackgroundColor)
+	private static Color? GetBackgroundColor(Brush background)
 	{
-		contentBackgroundColor = null;
-		if (TryGetContentBackground(out var background))
+		return background switch
 		{
-			if (background is not SolidColorBrush backgroundColorBrush)
-			{
-				throw new NotSupportedException("[ShadowContainer] Unsupported Background brush: when using inner shadows the only supported brush type for the Background property is SolidBrushColor");
-			}
-
-			if (backgroundColorBrush.Color != Color.FromArgb(0, 0, 0, 0))
-			{
-				contentBackgroundColor = backgroundColorBrush.Color;
-			}
-
-			TrySetContentBackground(new SolidColorBrush(Color.FromArgb(0, 0, 0, 0)));
-			return true;
-		}
+			SolidColorBrush scb => scb.Color,
 
-		return false;
+			null => default(Color?),
+			_ => throw new NotSupportedException($"Invalid background brush type: {background.GetType().Name}")
```

Comment:
it is probably VERY BAD to throw inside a paint procedure of skia
we can probably just treat it as transparent and continue as is
and, maybe just log at err-level here
\---
we are already throwing during the setup with `BindToPaintingProperties`